# MateFacil

<img src="../../pics/cover.png" width=600> 

https://www.youtube.com/@MateFacilYT

## Curso Completo de Aritmética

<img src="../../pics/cover_curso_aritmetica.png" width=400> 

https://www.youtube.com/playlist?list=PL9SnRnlzoyX1MVuXSqPt2Q_gxC8RGcIu_

In [1]:
pwd

'c:\\Users\\crodr\\Tutoriales\\Math\\MateFacil\\Aritmetica_basica\\Curso_Aritmetica'

In [2]:
from astropy import units as u
import sympy as sp
from IPython.display import display, Math

a, m, x, y, z = sp.symbols("a, m, x, y, z", real=True)

spacing = r"\,= \,"

## Video 154

Cómo convertir una fracción a número decimal 

$$\frac{3}{5}$$

In [ ]:
# expr_raw = r"\frac{3}{5}"
frac = sp.Rational(3, 5)
resultado = frac.n(1)

# display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)} {spacing} {resultado}"))
display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

Cómo convertir una fracción a número decimal 

$$\frac{7}{2}$$

In [ ]:
frac = sp.Rational(7, 2)
resultado = frac.n(2)

display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

## Video 155

Cómo convertir una fracción a número decimal 

$$\frac{3}{8}$$

In [ ]:
frac = sp.Rational(3, 8)
resultado = frac.n(3)


display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

Cómo convertir una fracción a número decimal 

$$\frac{9}{20}$$

In [ ]:
frac = sp.Rational(9, 20)
resultado = frac.n(2)
# resultado = frac.evalf()


display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

## Video 156

Cómo convertir una fracción a número decimal 

$$\frac{4}{3}$$

In [ ]:
frac = sp.Rational(4, 3)
resultado = frac.evalf()

display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

Cómo convertir una fracción a número decimal 

$$\frac{1}{7}$$

In [ ]:
frac = sp.Rational(1, 7)
resultado = frac.evalf()

display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

## Video 157

Cómo convertir una fracción a número decimal 

$$\frac{2}{9}$$

In [ ]:
frac = sp.Rational(2, 9)
resultado = frac.evalf()

display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

Cómo convertir una fracción a número decimal 

$$\frac{19}{90}$$

In [ ]:
frac = sp.Rational(19, 90)
resultado = frac.evalf()

display(Math(rf"{sp.latex(frac)} {spacing} {resultado}"))


<IPython.core.display.Math object>

## Video 159

Cómo convertir decimales finitos a fracciones 

$$0.8$$

In [27]:
expr_raw = r"0.8"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$0.25$$

In [28]:
expr_raw = r"0.25"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$0.244$$

In [29]:
expr_raw = r"0.244"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$3.5$$

In [30]:
expr_raw = r"3.5"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$6.13$$

In [31]:
expr_raw = r"6.13"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

## Video 160

Cómo convertir decimales finitos a fracciones 

$$3.6$$

In [32]:
expr_raw = r"3.6"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$2.22$$

In [33]:
expr_raw = r"2.22"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales finitos a fracciones 

$$0.125$$

In [34]:
expr_raw = r"0.125"
frac = sp.Rational(expr_raw)

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

## Video 161

Cómo convertir decimales infinitos a fracciones:

* Decimales periódicos puros. Toda la parte fraccionaria se repite.

$$0.\overline{3}$$


In [5]:
expr_raw = r"0.\overline{3}"
frac = sp.sympify("0.[3]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>


* Decimales periódicos mixtos. Sólo se repiten los últimos digitos de la parte fraccionaria.

$$0.\overline{12}$$

In [6]:
expr_raw = r"0.\overline{12}"
frac = sp.sympify("0.[12]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>


* Decimales periódicos mixtos. Sólo se repiten los últimos digitos de la parte fraccionaria.

$$5.\overline{18}$$

In [7]:
expr_raw = r"5.\overline{18}"
frac = sp.sympify("5.[18]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

## Video 162

Cómo convertir decimales periódicos puros a fracciones

$$0.\overline{123}$$


In [8]:
expr_raw = r"0.\overline{123}"
frac = sp.sympify("0.[123]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))


<IPython.core.display.Math object>

Cómo convertir decimales periódicos puros a fracciones

$$2.\overline{27}$$

In [9]:
expr_raw = r"2.\overline{27}"
frac = sp.sympify("2.[27]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

Cómo convertir decimales periódicos puros a fracciones

$$3.\overline{03}$$

In [10]:
expr_raw = r"3.\overline{03}"
frac = sp.sympify("3.[03]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

## Video 163

Cómo convertir decimales periódicos mixtos a fracciones

$$0.2\overline{6}$$


In [11]:
expr_raw = r"0.2\overline{6}"
frac = sp.sympify("0.2[6]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

Cómo convertir decimales periódicos mixtos a fracciones

$$5.32\overline{45}$$

In [12]:
expr_raw = r"5.32\overline{45}"
frac = sp.sympify("5.32[45]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

## Video 164

Cómo convertir decimales periódicos mixtos a fracciones

$$1.1\overline{3}$$


In [14]:
expr_raw = r"1.1\overline{3}"
frac = sp.sympify("1.1[3]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

Cómo convertir decimales periódicos mixtos a fracciones

$$0.13\overline{24}$$


In [16]:
expr_raw = r"0.13\overline{24}"
frac = sp.sympify("0.13[24]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

Cómo convertir decimales periódicos mixtos a fracciones

$$0.002\overline{54}$$


In [18]:
expr_raw = r"0.002\overline{54}"
frac = sp.sympify("0.002[54]")

display(Math(rf"{expr_raw} {spacing} {sp.latex(frac)}"))

<IPython.core.display.Math object>

## Video 166

Cómo redondear números

Redondear al entero más próximo

$$12.002$$


In [19]:
num = 12.002
num_int = f"{num:.0f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_int)}"))

<IPython.core.display.Math object>

Redondear al entero más próximo

$$67.772$$


In [25]:
num = 67.772
num_int = f"{num:.0f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_int)}"))

<IPython.core.display.Math object>

Redondear al entero más próximo

$$3.888$$


In [ ]:
num = 3.888
num_int = f"{num:.0f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_int)}"))

<IPython.core.display.Math object>

Redondear al centésimo más próximo

$$0.3468$$


In [28]:
num = 0.3468
num_cent = f"{num:.2f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_cent)}"))

<IPython.core.display.Math object>

Redondear al centésimo más próximo

$$7.6666$$


In [29]:
num = 7.6666
num_cent = f"{num:.2f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_cent)}"))

<IPython.core.display.Math object>

Redondear al centésimo más próximo

$$12.72109$$


In [3]:
num = 12.72109
num_cent = f"{num:.2f}"
display(Math(rf" {sp.latex(num)} {spacing} {sp.latex(num_cent)}"))

<IPython.core.display.Math object>

Redondear al millar más próximo

$$12\; 998\; 662$$


In [47]:
def fmt(n):
    return f"{int(n):,}".replace(
        ",", r"\;"
    )  # LaTeX usa {,} como separador, cambiar por un espacio en blanco


In [6]:
num = sp.Integer(12998662)
num_millar = (num / 1000).round() * 1000

display(Math(rf"{fmt(num)} \sim {fmt(num_millar)}"))


<IPython.core.display.Math object>

Redondear al millar más próximo

$$9\; 345\; 121$$


In [7]:
num = sp.Integer(9345121)
num_millar = (num / 1000).round() * 1000

display(Math(rf"{fmt(num)} \sim {fmt(num_millar)}"))

<IPython.core.display.Math object>

Redondear al millar más próximo

$$56\; 998$$


In [8]:
num = sp.Integer(56998)
num_millar = (num / 1000).round() * 1000

display(Math(rf"{fmt(num)} \sim {fmt(num_millar)}"))

<IPython.core.display.Math object>

## Video 167

Notación científica


In [ ]:
spacing = r"\,= \,"


def fmt(n):
    return f"{int(n):,}".replace(
        ",", r"\;"
    )  # LaTeX usa {,} como separador, cambiar por un espacio en blanco

In [ ]:
expr_raw = r"9 \times 10^2"
expr_sp = sp.Integer(9e2)
display(Math(rf"{expr_raw} {spacing}  {sp.latex(expr_sp)}"))

<IPython.core.display.Math object>

In [ ]:
expr_raw = r"13 \times 10^6"
expr_sp = sp.Integer(13e6)
display(Math(rf"{expr_raw} {spacing}  {fmt(expr_sp)}"))

<IPython.core.display.Math object>

In [57]:
expr_raw = r"4.89 \times 10^4"
expr_sp = 4.89e4
display(Math(rf"{expr_raw} {spacing}  {fmt(expr_sp)}"))

<IPython.core.display.Math object>

In [68]:
expr_raw = r"3.995 \times 10^{10}"
expr_sp = 3.995e10
display(Math(rf"{expr_raw} {spacing}  {fmt(expr_sp)}"))

<IPython.core.display.Math object>

In [ ]:
spacing = r"\, \approx \,"


def to_latex_sci(num, decimals=2):
    mantissa, exp = f"{num:.{decimals}e}".split("e")
    exp = int(exp)
    return rf"{mantissa} \times 10^{{{exp}}}"

In [80]:
num = 12_000
num_sci_latex = to_latex_sci(num, decimals=1)

display(Math(rf"{fmt(num)} {spacing}  {num_sci_latex}"))


<IPython.core.display.Math object>

In [75]:
num = 667_000_000
num_sci_latex = to_latex_sci(num)

display(Math(rf"{fmt(num)} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

In [81]:
num = 1_677_995_000
num_sci_latex = to_latex_sci(num)

display(Math(rf"{fmt(num)} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

In [82]:
num = 23_300_766_192_040
num_sci_latex = to_latex_sci(num)

display(Math(rf"{fmt(num)} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

## Video 168

Notación científica para números pequeños


In [31]:
spacing = r"\, = \,"

expr_raw = r"9 \times 10^{-2}"
expr_sp = sp.Float(9e-2)
display(Math(rf"{expr_raw} {spacing}  {sp.latex(expr_sp)}"))

<IPython.core.display.Math object>

In [32]:
expr_raw = r"13.95 \times 10^{-3}"
expr_sp = sp.Float(13.95e-3)
display(Math(rf"{expr_raw} {spacing}  {sp.latex(expr_sp)}"))

<IPython.core.display.Math object>

In [33]:
expr_raw = r"0.014 \times 10^{-5}"
expr_sp = sp.Float(0.014e-5)
resultado_expandido = f"{float(expr_sp):.8f}"
display(Math(rf"{expr_raw} {spacing}  {sp.latex(resultado_expandido)}"))

<IPython.core.display.Math object>

In [ ]:
def to_latex_sci(num, decimals=2):
    mantissa, exp = f"{num:.{decimals}e}".split("e")
    exp = int(exp)
    return rf"{mantissa} \times 10^{{{exp}}}"


In [26]:
spacing = r"\, = \,"
expr_raw = r"0.000043"

num = 0.000043
num_sci_latex = to_latex_sci(num, 1)

display(Math(rf"{expr_raw} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

In [30]:
expr_raw = r"0.000000862"

num = 0.000000862
num_sci_latex = to_latex_sci(num)

display(Math(rf"{expr_raw} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

In [34]:
expr_raw = r"0.000000331966411"
spacing = r"\, \approx \,"
num = 0.000000331966411
num_sci_latex = to_latex_sci(num)

display(Math(rf"{expr_raw} {spacing}  {num_sci_latex}"))

<IPython.core.display.Math object>

## Video 169

Cambiar exponente de la potencia de 10


In [43]:
spacing = r"\, = \,"


def adaptar_exponente_latex(numero, exponente_objetivo):
    # Calculamos el nuevo coeficiente dividiendo el número total entre la potencia deseada
    nuevo_coeficiente = numero / (10**exponente_objetivo)

    # Lo convertimos a float de Python para limpiar ceros innecesarios al final
    coef_float = float(nuevo_coeficiente)

    # Si el resultado es un número entero para Python (ej: 4500.0), quitamos el decimal .0
    if coef_float.is_integer():
        coef_str = str(int(coef_float))
    else:
        coef_str = str(coef_float)

    return rf"{coef_str} \times 10^{{{exponente_objetivo}}}"

Cambiar exponente de la potencia de 10

$$450 \times 10^3 = \_\_\_\_ \times 10^5 = \_\_\_\_ \times 10^2 = \_\_\_\_ \times 10^{-2}$$


In [ ]:
expr_raw = r"450 \times 10^3"
num = sp.Float(450e3)
res_1_exp5 = adaptar_exponente_latex(num, 5)
res_2_exp2 = adaptar_exponente_latex(num, 2)
res_3_exp_neg2 = adaptar_exponente_latex(num, -2)

display(
    Math(
        rf"{expr_raw} {spacing}  {res_1_exp5} {spacing}  {(res_2_exp2)} {spacing}  {res_3_exp_neg2}"
    )
)


<IPython.core.display.Math object>

Cambiar exponente de la potencia de 10

$$21.995 \times 10^6 = \_\_\_\_ \times 10^2 = \_\_\_\_ \times 10^{-1} = \_\_\_\_ \times 10^9$$


In [ ]:
expr_raw = r"21.995 \times 10^6"
num = sp.Float(21.995e6)
res_1_exp2 = adaptar_exponente_latex(num, 2)
res_2_exp_neg1 = adaptar_exponente_latex(num, -1)
res_3_exp9 = adaptar_exponente_latex(num, 9)

display(
    Math(
        rf"{expr_raw} {spacing}  {res_1_exp2} {spacing}  {(res_2_exp_neg1)} {spacing}  {res_3_exp9}"
    )
)

<IPython.core.display.Math object>

## Video 170

Cambiar exponente de la potencia de 10

$$12.51 \times 10^{-5} = \_\_\_\_ \times 10^{-3} = \_\_\_\_ \times 10^{-8} = \_\_\_\_ \times 10^2$$

In [ ]:
def adaptar_exponente_latex(numero, exponente_objetivo):
    # 1. Calculamos el nuevo coeficiente usando la precisión de SymPy
    nuevo_coeficiente = numero / (10**exponente_objetivo)
    coef_float = float(nuevo_coeficiente)

    # 2. Si es entero, limpiamos el .0
    if coef_float.is_integer():
        coef_str = str(int(coef_float))
    else:
        # LA CLAVE: Forzamos a Python a redondear directamente en la cadena de texto.
        # Usamos :.12f para limpiar la basura binaria, y luego quitamos los ceros sobrantes.
        coef_str = f"{coef_float:.12f}".rstrip("0").rstrip(".")

    return rf"{coef_str} \times 10^{{{exponente_objetivo}}}"

In [ ]:
expr_raw = r"12.51 \times 10^{-5}"
num = sp.Float(12.51e-5)
res_1_neg3 = adaptar_exponente_latex(num, -3)
res_2_neg8 = adaptar_exponente_latex(num, -8)
res_3_exp2 = adaptar_exponente_latex(num, 2)

display(
    Math(
        rf"{expr_raw} {spacing}  {res_1_neg3} {spacing}  {(res_2_neg8)} {spacing}  {res_3_exp2}"
    )
)

<IPython.core.display.Math object>

## Video 173

### Regla de tres compuesta

Se emplean 12 hombres durante 6 días para cavar una zanja de 30m de largo, 8m de ancho y 4m de alto, trabajando 6 horas diarias. Si se emplea el doble número de hombres durante 5 días, para cavar otra zanja de 20m de largo, 12m de ancho y 3m de alto, ¿cuántas horas diarias deben trabajar?

* Dos cantidades son `directamente proporcionales`, si al aumentar una aumenta también la otra $\mp$

    * Si aumenta el largo de la zanja, aumentará el número de horas de trabajo

* Dos cantidades son `inversamente proporcionales`, si al aumentar una disminuye la otra $\pm$

    * Si aumenta el número de hombres, disminuirá el número de horas de trabajo

#### Solución:

* Paso 1. Escribimos las cantidades, de forma ordenada

* Paso 2. Colocar los signos

<br>

<div align="center">

| + | + | - | - | - | + |
| :---: | :---: | :---: | :---: | :---: | :---: |
| **Hombres** | **Días** | **Largo** | **Ancho** | **Alto** | **Horas** |
| 12 | 6 | 30 | 8 | 4 | 6 |
| 24 | 5 | 20 | 12 | 3 | $x$ |
| - | - | + | + | + | - |

</div>

* Paso 3. Multiplicar todos los + y dividir entre todos los -

$$x = \frac{12 \times 6 \times 20 \times 12 \times 3 \times 6}{24 \times 5 \times 30 \times 8 \times 4}$$



In [4]:
expr_raw = r"\frac{12 \times 6 \times 20 \times 12 \times 3 \times 6}{24 \times 5 \times 30 \times 8 \times 4}"
x = expr_sp = (12 * 6 * 20 * 12 * 3 * 6) / (24 * 5 * 30 * 8 * 4)

display(Math(rf"x = {expr_raw} {spacing} {x} \text{{ horas al día}}"))


<IPython.core.display.Math object>

In [5]:
# Convertimos a string y separamos por el punto decimal
str_entera, str_decimal = str(x).split(".")

horas = int(str_entera)
parte_decimal = float(f"0.{str_decimal}")
minutos = int(parte_decimal * 60)

display(
    Math(
        rf" {x} \text{{ horas al día}} {spacing} {horas} \text{{ horas con }} {minutos} \text{{ minutos al día}}"
    )
)


<IPython.core.display.Math object>

## Video 174

### Porcentajes y descuentos

a) Adrián compró un automóvil en \$320 000, el precio incluía entre seguro, impuestos y accesorios 25% mas, ¿cuál era el precio del automóvil sin contar seguro, impuestos y accesorios?


Precio auto:        100% $\implies x$

Precio cobrado:     125% $\implies 320 000$

$$x = \frac{100\% \; 320000}{125\%}$$


In [9]:
# Regla de tres
# Es conveniente convertir los porcentajes en tanto por uno
x = (1 * 320000) / (1.25)
print(f"Precio neto del coche: ${x:,.0f}")

Precio neto del coche: $256,000


## Video 175

### Porcentajes y descuentos

b) Paola compró una bicicleta de montaña en $1200, si el precio incluía una rebaja de 20%, ¿cuál era el precio normal de la bicicleta?

Precio compra:        80% $\implies \$1200$

Precio normal:     100% $\implies x$

$$x = \frac{100\% \; 1200}{80\%}$$


In [11]:
# Regla de tres
# Es conveniente convertir los porcentajes en tanto por uno
x = (1 * 1200) / (0.8)
print(f"Precio normal de la bicicleta: ${x:,.0f}")

Precio normal de la bicicleta: $1,500


## Video 176

### Divisivilidad

* Un número es divisible por 2 si es un número par.

* Un número es divisible por 3 si es 3, 6 o 9. 

O si la suma de sus digitos da como resultado 3, 6 o 9

* Un número es divisible por 5 si su último dígito es 0 o 5.

* Un número es divisible por 9 si la suma de sus dígitos es 9 o divisible por 9. 



## Video 178

### Mínimo Común Multiplo

In [ ]:
import math

# Obtener mcd de 15, 20
mcm = math.lcm(15, 20)
print(f"El mcm(15, 20) = {mcm}")

El mcm(15, 20) = 60


In [30]:
# Obtener mcd de 12, 15, 24
mcm = math.lcm(12, 15, 24)
print(f"El mcm(12, 15, 24) = {mcm}")

El mcm(12, 15, 24) = 120


In [31]:
# Obtener mcd de 12, 15
mcm = math.lcm(12, 15)
print(f"El mcm(12, 15) = {mcm}")

El mcm(12, 15) = 60


In [32]:
# Obtener mcd de 20, 25
mcm = math.lcm(20, 25)
print(f"El mcm(20, 25) = {mcm}")

El mcm(20, 25) = 100


In [33]:
# Obtener mcd de 4, 6, 10
mcm = math.lcm(4, 6, 10)
print(f"El mcm(4, 6, 10) = {mcm}")

El mcm(4, 6, 10) = 60


In [34]:
# Obtener mcd de 12, 16, 20
mcm = math.lcm(12, 16, 20)
print(f"El mcm(12, 16, 20) = {mcm}")

El mcm(12, 16, 20) = 240


## Video 183

### Exponente fraccionado

$$9^{\frac{3}{2}}$$



In [ ]:
expr_raw = r"9^{\frac{3}{2}}"
expr_sp = 9 ** sp.Rational(3, 2)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>


### Exponente fraccionado

$$8^{\frac{2}{3}}$$

In [ ]:
expr_raw = r"8^{\frac{2}{3}}"
expr_sp = 8 ** sp.Rational(2, 3)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>

### Exponente fraccionado

$$100^{\frac{3}{2}}$$

In [ ]:
expr_raw = r"100^{\frac{3}{2}}"
expr_sp = 100 ** sp.Rational(3, 2)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>


### Exponente fraccionado

$$64^{\frac{2}{3}}$$

In [ ]:
expr_raw = r"64^{\frac{2}{3}}"
expr_sp = 64 ** sp.Rational(2, 3)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>


### Exponente fraccionado

$$27^{\frac{4}{3}}$$

In [ ]:
expr_raw = r"27^{\frac{4}{3}}"
expr_sp = 27 ** sp.Rational(4, 3)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>


### Exponente fraccionado

$$49^{\frac{3}{2}}$$

In [ ]:
expr_raw = r"49^{\frac{3}{2}}"
expr_sp = 49 ** sp.Rational(3, 2)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>


### Exponente fraccionado

$$2013^{\frac{0}{5}}$$

In [ ]:
expr_raw = r"2013^{\frac{0}{5}}"
expr_sp = 2013 ** sp.Rational(0, 5)

spacing = r"\,= \,"
display(Math(rf"{expr_raw} {spacing} {sp.latex(expr_sp)} "))

<IPython.core.display.Math object>

## Video 186

### Criba de Eratóstenes





In [ ]:
def criba_eratostenes(n):
    """Encuentra todos los números primos hasta n usando la Criba de Eratóstenes."""
    if n < 2:
        return []

    es_primo = [True] * (n + 1)
    es_primo[0] = es_primo[1] = False

    p = 2
    while p * p <= n:
        if es_primo[p]:
            for multiplo in range(p * p, n + 1, p):
                es_primo[multiplo] = False
        p += 1

    return [i for i, primo in enumerate(es_primo) if primo]


In [16]:
limite = 50
primos = criba_eratostenes(limite)
print(f"Primos hasta {limite}: {primos}")
print(f"Total: {len(primos)} números primos")

Primos hasta 50: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]
Total: 15 números primos


In [17]:
limite = 100
primos = criba_eratostenes(limite)
print(f"Primos hasta {limite}: {primos}")
print(f"Total: {len(primos)} números primos")

Primos hasta 100: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]
Total: 25 números primos


In [18]:
limite = 200
primos = criba_eratostenes(limite)
print(f"Primos hasta {limite}: {primos}")
print(f"Total: {len(primos)} números primos")

Primos hasta 200: [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199]
Total: 46 números primos


## Video 187

### Máximo Común Divisor


In [ ]:
# Obtener mcd de 30 y 45
# De la librería estándar math usamos gcd (greatest common divisor)
import math

mcd = math.gcd(30, 45)
print(f"El mcd(30, 45) = {mcd}")

El mcd(30, 45) = 15


In [21]:
# Obtener mcd de 18, 24, 36
mcd = math.gcd(18, 24, 36)
print(f"El mcd(18, 24, 36) = {mcd}")

El mcd(18, 24, 36) = 6


In [22]:
# Obtener mcd de 21, 30
mcd = math.gcd(21, 30)
print(f"El mcd(21, 30) = {mcd}")

El mcd(21, 30) = 3


In [23]:
# Obtener mcd de 8, 16, 32
mcd = math.gcd(8, 16, 32)
print(f"El mcd(8, 16, 32) = {mcd}")

El mcd(8, 16, 32) = 8


In [24]:
# Obtener mcd de 30, 60, 90
mcd = math.gcd(30, 60, 90)
print(f"El mcd(30, 60, 90) = {mcd}")

El mcd(30, 60, 90) = 30


## Video 188

### Fracciones con exponente

$$\left( \frac{3}{4} \right)^2 = \frac{3^4}{4^4} = \frac{3 \times 3}{4 \times 4} = \frac{9}{16}$$


In [ ]:
expr_raw = r"\left( \frac{3}{4} \right)^2"
resultado = sp.Rational(3, 4) ** 2

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{3}{5} \right)^3 $$

In [ ]:
expr_raw = r"\left( \frac{3}{5} \right)^3"
resultado = sp.Rational(3, 5) ** 3

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{1}{2} \right)^4 $$

In [ ]:
expr_raw = r"\left( \frac{1}{2} \right)^4 "
resultado = sp.Rational(1, 2) ** 4

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{9}{21} \right)^0 $$

In [ ]:
expr_raw = r"\left( \frac{9}{21} \right)^0"
resultado = sp.Rational(9, 21) ** 0

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{7}{10} \right)^2 $$

In [ ]:
expr_raw = r"\left( \frac{7}{10} \right)^2"
resultado = sp.Rational(7, 10) ** 2

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{8}{2} \right)^2 $$

In [ ]:
expr_raw = r"\left( \frac{8}{2} \right)^2"
resultado = sp.Rational(8, 2) ** 2

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{11}{100} \right)^0 $$

In [ ]:
expr_raw = r"\left( \frac{11}{100} \right)^0 "
resultado = sp.Rational(11, 100) ** 0

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{3}{4} \right)^3 $$

In [ ]:
expr_raw = r"\left( \frac{3}{4} \right)^3"
resultado = sp.Rational(3, 4) ** 3

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$ \left( \frac{1}{10} \right)^4 $$

In [ ]:
expr_raw = r"\left( \frac{1}{10} \right)^4"
resultado = sp.Rational(1, 10) ** 4

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

$$\left( \frac{5}{5} \right)^0 $$

In [ ]:
expr_raw = r"\left( \frac{5}{7} \right)^0"
resultado = sp.Rational(5, 7) ** 0

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 188

### Fracciones con exponente negativo

$$\left( \frac{2}{3} \right)^{-3} = \left( \frac{3}{2} \right)^{3} = \frac{3^3}{2^3} 
= \frac{3 \times 3 \times 3}{2 \times 2 \times 2} = \frac{27}{8}$$



In [4]:
expr_raw = r"\left( \frac{2}{3} \right)^{-3} "
resultado = sp.Rational(2, 3) ** -3

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Fracciones con exponente negativo

$$\left( \frac{5}{2} \right)^{-2} $$



In [5]:
expr_raw = r"\left( \frac{5}{2} \right)^{-2} "
resultado = sp.Rational(5, 2) ** -2

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Fracciones con exponente negativo

$$\left( \frac{4}{3} \right)^{-1} $$



In [6]:
expr_raw = r"\left( \frac{4}{3} \right)^{-1} "
resultado = sp.Rational(4, 3) ** -1

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 190

### Fracciones con exponente fraccionario

$$\left( \frac{9}{4} \right)^{\frac{1}{2}} = \frac {9^{\frac{1}{2}}}{4^{\frac{1}{2}}} 
= \frac {\sqrt{9}}{\sqrt{4}} = \frac {3}{2}$$



In [ ]:
expr_raw = r"\left( \frac{9}{4} \right)^{\frac{1}{2}} "
resultado = sp.Rational(9, 4) ** (sp.Rational(1, 2))

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Fracciones con exponente fraccionario

$$\left( \frac{36}{16} \right)^{\frac{1}{2}} $$



In [ ]:
expr_raw = r" \left( \frac{36}{16} \right)^{\frac{1}{2}}"
resultado = sp.Rational(36, 16) ** (sp.Rational(1, 2))

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))


<IPython.core.display.Math object>

### Fracciones con exponente fraccionario

$$\left( \frac{8}{27} \right)^{\frac{2}{3}} $$



In [ ]:
expr_raw = r" \left( \frac{8}{27} \right)^{\frac{2}{3}} "
resultado = sp.Rational(8, 27) ** (sp.Rational(2, 3))

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 191

### Cómo sacar séptima

1. Se quita el último dígito al número y se duplica dicho dígito

2. Al número que queda después de haberle quitado el último dígito se le resta el número obtenido en el paso 1


Ejemplo:

¿Es 91 divisible por 7?

$$91 \implies 1 \times 2 = 2$$

$$9 - 2 = 7$$

El resultado 7 es divisible por 7, luego 91 es divisible por 7





In [16]:
expr_raw = r" 91 \div 7"
resultado = sp.Integer(91) / sp.Integer(7)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

#### Es divisible por 7?

$$455 \div 7$$


In [17]:
expr_raw = r" 455 \div 7"
resultado = sp.Integer(455) / sp.Integer(7)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

#### Es divisible por 7?

$$834 \div 7$$


In [18]:
expr_raw = r" 834 \div 7"
resultado = sp.Integer(834) / sp.Integer(7)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 192

### Criterio de divisibilidad por 11

1. Se suman los digitos que están en posición par

2. Se suman los digitos que están en posición impar

3. Se restan los resultados obtenidos en los pasos 2 y 3

Sí el resultado es cero o divisible por 11, entonces el número en cuestión también lo es.

Ejemplo:

¿Es 143 divisible por 11?

$$\text{Sumamos los impares }143 \implies 1 + 3 = 4$$

$$\text{Sumamos los pares }143 \implies 4 + . = 4$$

$$\text{Restamos resultados 1 - 2 } \implies 4 - 4 = 0$$


El resultado $0$ es divisible por $11$ pues cualquier número multiplicado por $0$ nos da $0$, luego el número $143$ es divisible por $11$

$$143 \div 11 = 13$$





### Criterio de divisibilidad por 11

$$2827 \div 11$$


In [ ]:
expr_raw = r" 2827 \div 11"
resultado = sp.Integer(2827) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Criterio de divisibilidad por 11

$$2681 \div 11$$


In [20]:
expr_raw = r" 2681 \div 11"
resultado = sp.Integer(2681) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Criterio de divisibilidad por 11

$$123321 \div 11$$


In [21]:
expr_raw = r" 123321 \div 11"
resultado = sp.Integer(123321) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Criterio de divisibilidad por 11

$$12100 \div 11$$


In [22]:
expr_raw = r" 12100 \div 11"
resultado = sp.Integer(12100) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Criterio de divisibilidad por 11

$$5555 \div 11$$


In [23]:
expr_raw = r" 5555 \div 11"
resultado = sp.Integer(5555) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

### Criterio de divisibilidad por 11

$$5623 \div 11$$


In [24]:
expr_raw = r" 5623 \div 11"
resultado = sp.Integer(5623) / sp.Integer(11)

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 194

### Reglas para la suma y resta

1. Los números con signos iguales se suman y el resultado tendrá el mismo signo que los números dados

2. Los números con signos opuestos se restan y el resultado tendrá el mismo signo que el número mayor

#### Ejemplos

$$-1 -3$$


In [5]:
expr_raw = r"-1 -3"
resultado = -1 - 3

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

#### Ejemplos

$$-5 +6$$


In [ ]:
expr_raw = r"-5 +6"
resultado = -5 + 6

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

#### Ejemplos

$$-2 -2$$


In [ ]:
expr_raw = r"-2 -2"
resultado = -2 - 2

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

#### Ejemplos

$$3 -4$$


In [ ]:
expr_raw = r"3 -4"
resultado = 3 - 4

display(Math(rf"{expr_raw} {spacing} {sp.latex(resultado)}"))

<IPython.core.display.Math object>

## Video 199

### Convertir fracción a decimal y porcentaje

$$\frac{3}{4}$$


In [22]:
expr_raw = r"\frac{3}{4}"
frac = sp.Rational(3, 4)
decimal = frac.evalf()
porcentaje = int(decimal * 100)

display(Math(rf"{expr_raw} {spacing} {sp.latex(decimal)} = {sp.latex(porcentaje)}\%"))

<IPython.core.display.Math object>

### Convertir porcentaje a decimal y a fracción 

$$45\%$$

In [19]:
expr_raw = r"45\%"
porcentaje = 45
decimal = porcentaje / 100
frac = sp.Rational(porcentaje, 100)

display(Math(rf"{expr_raw} {spacing} {sp.latex(decimal)} = {sp.latex(frac)}"))

<IPython.core.display.Math object>

### Convertir fracción a decimal y porcentaje

$$\frac{3}{5}$$


In [21]:
expr_raw = r"\frac{3}{5}"
frac = sp.Rational(3, 5)
decimal = frac.evalf()
porcentaje = int(decimal * 100)

display(Math(rf"{expr_raw} {spacing} {sp.latex(decimal)} = {sp.latex(porcentaje)}\%"))

<IPython.core.display.Math object>

### Convertir porcentaje a decimal y a fracción 

$$25\%$$

In [23]:
expr_raw = r"25\%"
porcentaje = 25
decimal = porcentaje / 100
frac = sp.Rational(porcentaje, 100)

display(Math(rf"{expr_raw} {spacing} {sp.latex(decimal)} = {sp.latex(frac)}"))

<IPython.core.display.Math object>